subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r60/ToolingApiPropertiesLoaderCrossVersionSpec.groovy

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.tooling.r60
+
+
+import org.gradle.integtests.tooling.fixture.TargetGradleVersion
+import org.gradle.integtests.tooling.fixture.TestOutputStream
+import org.gradle.integtests.tooling.fixture.ToolingApiSpecification
+import org.gradle.integtests.tooling.fixture.ToolingApiVersion
+import spock.lang.Issue
+
+@ToolingApiVersion('>=6.0')
+@TargetGradleVersion('>=6.0')
```

Comment:
Did you try different versions of the client or provider here to see when the behavior changed?

e.g., does a 5.6 TAPI client working with a 6.0 provider get the right behavior? I think yes.
And does a 6.0 TAPI client working with a 5.6 provider get the right behavior? I think no.

Reply:
Your assumption is correct, only using the Gradle 5.6 client breaks the test, but still works with older TAPI versions. I've lowered the required TAPI version to 3.0.